cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -556,6 +560,11 @@ var (
 			rawUnsetPaths, _ := cmd.Flags().GetStringSlice("unset")
 			unsetPaths := util.NormalizePaths(rawUnsetPaths)
 
+			var device ttnpb.EndDevice
+			if updateDeviceLocation(&device, cmd.Flags()) {
+				paths = append(paths, "locations")
+			}
```

Comment:
This wouldn't correctly perform updates.

If any of the location flags is set, you need to fetch `locations` from the Identity Server and then update using the locations in `existingDevice`.